src/util/sawyercoding.c

```diff
@@ -320,8 +320,10 @@ static size_t decode_chunk_repeat(uint8 *buffer, size_t length)
 		} else {
 			count = (src[i] & 7) + 1;
 			copyOffset = dst + (int)(src[i] >> 3) - 32;
-			for (j = 0; j < count; j++)
+			assert(dst + count < buffer + length);
```

Comment:
This isn't right, length is the compressed length, not the uncompressed length. The buffer is a block of compressed data which expands on itself to something bigger. There is no way of knowing what the uncompressed data length is ahead of time.